packages/firestore/src/core/firestore_client.ts

```diff
@@ -415,6 +418,20 @@ export class FirestoreClient {
     );
   }
 
+  async getDocumentViaSnapshotListener(
```

Comment:
Optional: consider something like `getSingleDocumentViaSnapshotListener`. As it stands, these function names are really similar to each other.

Reply:
This is actually the name that was used previously and matches other SDKs, such as Android. I am leaving as is for consistency.